src/com/amazon/corretto/crypto/provider/EdGen.java

```diff
@@ -35,10 +40,11 @@ public void initialize(final int keysize, final SecureRandom random) {
 
   @Override
   public KeyPair generateKeyPair() {
-    final EvpEdPrivateKey privateKey;
-    final EvpEdPublicKey publicKey;
-    privateKey = new EvpEdPrivateKey(generateEvpEdKey());
-    publicKey = privateKey.getPublicKey();
+    final EvpEdPrivateKey privateKey = new EvpEdPrivateKey(generateEvpEdKey());
+    final EvpEdPublicKey publicKey = privateKey.getPublicKey();
+    if (kf == null) { // This case indicates JDK EdDSA support conditions as described in the ctor
```

Comment:
Nit, can you use the full word. "constructor" not "ctor".

Reply:
sure.